crates/bevy_pbr/src/prepass/mod.rs

```diff
@@ -281,39 +355,55 @@ where
 
         let vertex_buffer_layout = layout.get_layout(&vertex_attributes)?;
 
-        // The fragment shader is only used when the normal prepass is enabled
-        // or the material uses alpha cutoff values and doesn't rely on the standard prepass shader
-        let fragment = if key.mesh_key.contains(MeshPipelineKey::NORMAL_PREPASS)
-            || ((key.mesh_key.contains(MeshPipelineKey::ALPHA_MASK)
-                || blend_key == MeshPipelineKey::BLEND_PREMULTIPLIED_ALPHA)
-                && self.material_fragment_shader.is_some())
-        {
-            // Use the fragment shader from the material if present
-            let frag_shader_handle = if let Some(handle) = &self.material_fragment_shader {
-                handle.clone()
-            } else {
-                PREPASS_SHADER_HANDLE.typed::<Shader>()
-            };
+        // Setup prepass fragment targets - normals in slot 0 (or None if not needed), motion vectors in slot 1
+        let mut targets = vec![];
+        targets.push(
+            key.mesh_key
+                .contains(MeshPipelineKey::NORMAL_PREPASS)
+                .then_some(ColorTargetState {
+                    format: NORMAL_PREPASS_FORMAT,
+                    blend: Some(BlendState::REPLACE),
+                    write_mask: ColorWrites::ALL,
+                }),
+        );
 
-            let mut targets = vec![];
-            // When the normal prepass is enabled we need a target to be able to write to it.
-            if key.mesh_key.contains(MeshPipelineKey::NORMAL_PREPASS) {
-                targets.push(Some(ColorTargetState {
-                    format: TextureFormat::Rgb10a2Unorm,
+        targets.push(
+            key.mesh_key
+                .contains(MeshPipelineKey::MOTION_VECTOR_PREPASS)
+                .then_some(ColorTargetState {
+                    format: MOTION_VECTOR_PREPASS_FORMAT,
                     blend: Some(BlendState::REPLACE),
                     write_mask: ColorWrites::ALL,
-                }));
-            }
+                }),
+        );
+
+        if targets.iter().all(Option::is_none) {
+            // if no targets are required then clear the list, so that no fragment shader is required
+            // (though one may still be used for discarding depth buffer writes)
+            targets.clear();
+        }
+
+        // The fragment shader is only used when the normal prepass or motion vectors prepass
+        // is enabled or the material uses alpha cutoff values
+        let fragment_required = key
+            .mesh_key
+            .intersects(MeshPipelineKey::ALPHA_MASK | MeshPipelineKey::BLEND_PREMULTIPLIED_ALPHA)
+            || !targets.is_empty();
```

Comment:
```suggestion 
        // is enabled or the material uses alpha cutoff values and doesn't rely on the standard 
        // prepass shader
        let fragment_required = !targets.is_empty()
            || ((key.mesh_key.contains(MeshPipelineKey::ALPHA_MASK)
                || blend_key == MeshPipelineKey::BLEND_PREMULTIPLIED_ALPHA)
                && self.material_fragment_shader.is_some())
```
This change would prevent the need for the dummy fragment shader in `prepass.wgsl`. It was included in #7878 but seems to got lost during integration into this PR.

I also believe the current code is misleading: `MeshPipelineKey::BLEND_PREMULTIPLIED_ALPHA` is a blend state and a single blend state could potentially consist of multiple set bits and two different blend states could intersect in their set bits (This isn't currently the case, but it could happen if more blend states would be added). This was the reason i used the comparison with `blend_key`.

Reply:
Also, I want to say I am sorry that my changes have lead to so many issues right before the release.

Reply:
not your fault at all, i mistakenly amended the check integrating it into this PR. I agree that we should amend the `.intersects(MeshPipelineKey::ALPHA_MASK | MeshPipelineKey::BLEND_PREMULTIPLIED_ALPHA)` to `((key.mesh_key.contains(MeshPipelineKey::ALPHA_MASK)
                || blend_key == MeshPipelineKey::BLEND_PREMULTIPLIED_ALPHA)`

the suggestion i made in the shadow fix PR to use a fragment_shader of `None` to check if we should include a fragment state was wrong as well, as None here implies that the shader should use the default prepass shader. So we do need to remove the `&& self.material_fragment_shader.is_some()`.

Reply:
Can you make a github suggestion for the last paragraph of what you said?

Reply:
Yes, `None` implies that the default prepass shader should be used. The idea of the `&& self.material_fragment_shader.is_some()` is that if the default prepass shader should be used, we only need the fragment shader if we have targets, because only then does the default prepass shader something useful.